datafusion/proto/src/to_proto.rs

```diff
@@ -639,17 +644,23 @@ impl TryFrom<&Expr> for protobuf::LogicalExprNode {
                         .collect::<Result<Vec<_>, Error>>()?,
                 })),
             },
-            Expr::AggregateUDF { fun, args } => Self {
-                expr_type: Some(ExprType::AggregateUdfExpr(
-                    protobuf::AggregateUdfExprNode {
-                        fun_name: fun.name.clone(),
-                        args: args.iter().map(|expr| expr.try_into()).collect::<Result<
-                            Vec<_>,
-                            Error,
-                        >>(
-                        )?,
-                    },
-                )),
+            Expr::AggregateUDF { fun, args, filter } => {
+                Self {
+                    expr_type: Some(ExprType::AggregateUdfExpr(
+                        Box::new(protobuf::AggregateUdfExprNode {
+                            fun_name: fun.name.clone(),
+                            args: args.iter().map(|expr| expr.try_into()).collect::<Result<
+                                Vec<_>,
+                                Error,
+                            >>(
+                            )?,
+                            filter: match filter {
+                                Some(e) => Some(Box::new(e.as_ref().try_into()?)),
+                                None => None,
+                            }
```

Comment:
I think you can write this kind of thing somewhat more concisely via

```suggestion
                            filter: filter
                              .map(|filter| Box::new(e.as_ref().try_into()))
                              .transpose()?
```

Reply:
I experimented with this locally but could not get this to work.